manifests/init.pp

```diff
@@ -6,6 +6,12 @@
 class git (
   $package        = $::git::params::package,
   $package_ensure = $::git::params::package_ensure,
+
+  ### START Hiera Lookups ###
+  $repo = hiera_has('git::repo', {}),
```

Comment:
While I don't mind adding a $repos variable, I'm not a fan of hiera_has (should that be hiera_hash?) here. Personally I don't use hiera, but wouldn't a $repos variable suffice here?